src/librustc_metadata/rmeta/mod.rs

```diff
@@ -286,6 +280,7 @@ define_tables! {
     mir: Table<DefIndex, Lazy!(mir::Body<'tcx>)>,
     promoted_mir: Table<DefIndex, Lazy!(IndexVec<mir::Promoted, mir::Body<'tcx>>)>,
     unused_generic_params: Table<DefIndex, Lazy<FiniteBitSet<u64>>>,
+    trait_impls: Table<u32, Lazy!(Table<DefIndex, Lazy!([DefIndex])>)>,
```

Comment:
We don't have sparse tables yet (I've been meaning to do that but haven't gotten around to it) so you should look at how much this change increases the size of `libcore-*.rlib` and `libstd-*.rlib` etc.

Reply:
You could maybe make this cheaper by removing the inner table and keeping a cache that's keyed on the `u32` with a value of `FxHashMap<DefIndex, Lazy<[DefIndex]>>` for every crate.

Also, the `u32`, should that be `CrateNum`? That would be clearer and as a `Table` it shouldn't cause problems I don't think.